src/main/java/com/kalessil/phpStorm/phpInspectionsEA/inspectors/codeSmell/NamingConvention/NamingConventionInspector.java

```diff
@@ -0,0 +1,49 @@
+package com.kalessil.phpStorm.phpInspectionsEA.inspectors.codeSmell.NamingConvention;
+
+import com.intellij.codeInspection.ProblemHighlightType;
+import com.intellij.codeInspection.ProblemsHolder;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiElementVisitor;
+import com.jetbrains.php.lang.psi.elements.PhpClass;
+import com.kalessil.phpStorm.phpInspectionsEA.openApi.BasePhpElementVisitor;
+import com.kalessil.phpStorm.phpInspectionsEA.openApi.BasePhpInspection;
+import com.kalessil.phpStorm.phpInspectionsEA.utils.NamedElementUtil;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Set;
+
+public class NamingConventionInspector extends BasePhpInspection {
+    private static final String message = "Class name does not follow naming convention";
+
+    public NamingRules rules = new NamingRules();
+
+
+    @NotNull
+    public String getShortName() {
+        return "NamingConventionInspection";
+    }
+
+    @NotNull
+    @Override
+    public PsiElementVisitor buildVisitor(@NotNull ProblemsHolder holder, boolean b) {
+        return new BasePhpElementVisitor() {
+            @Override
+            public void visitPhpClass(PhpClass phpClass) {
+                final PsiElement phpClassNameIdentifier = NamedElementUtil.getNameIdentifier(phpClass);
+                if (phpClassNameIdentifier != null) {
+                    // Create custom validate object to cache object type and extend list
+                    final PhpValidatableClass validatableClass = new PhpValidatableClass(phpClass);
+                    final Set<NamingRule> objectRules = rules.getRulesByType(validatableClass.getType());
+                    for (NamingRule rule : objectRules) {
+                        if (rule.isSupported(validatableClass) && !rule.isValid(phpClass)) {
+                            holder.registerProblem(phpClassNameIdentifier, message, ProblemHighlightType.WEAK_WARNING);
```

Comment:
1. Most simple. Provide regex + write class type. For example:
Trait name does not corresponding to the following regex `Trait$`
2. Try to detect what regexp math (end or start) 
  - `^MyTrait.+`   -  (Trait name should begin with `MyTrait` keyword
  -  `+Interface$` - Interface should end with `Interface` keyword
3. Add special name validation. Can be configured with dropdown:
 3.1. Start with
 3.2. End with
 3.3 Match regexp
In this case developer will configure rule and we can provide super informative message ;) 
Not so hard to implement =)